airqo-platform/AirQo-api
# null values in Events collection for AirQo devices

# Bugfix - Airqo device measurements null values

**_WHAT DOES THIS PR DO?_**
This partly addresses [Issue-273](https://github.com/airqo-platform/AirQo-api/issues/273). Attempts to resolve the first sub-issue about null values being inserted into the collections by the recently deployed Airqo data streaming function.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-460](https://airqoteam.atlassian.net/browse/PLAT-460?atlOrigin=eyJpIjoiY2E3NDI4MDRmYWNjNDMwNTgzY2I5N2Y2OGExZjQxMmUiLCJwIjoiaiJ9)


**_WHAT IS THE LINK TO THE PR BRANCH_**
[fix-airqo-null-values](https://github.com/airqo-platform/AirQo-api/tree/fix-airqo-null-values)

**_HOW DO I TEST OUT THIS PR?_**

To test out, run the commands below. Then observe  airqo events collection for null values.  (You can query data for one day using {"day": {$eq: ISODate('YYYY-MM-DD')}} eg {"day": {$eq: ISODate('2020-08-31')}} )

`cd AirQo-api/src/data-mgt/python/jobs`
`export AIRQO_API_BASE_URL = "{AIRQO_STAGING_BASE_URL}"`
`export FEEDS_BASE_URL = "{THINGS_SPEAK_BASE_URL}"`
`python airqo_device_measurements.py`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
N/A

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
[eliminate duplicate entries for days](https://github.com/airqo-platform/AirQo-api/pull/278)
